sql/2022/interop/popularity_of_contain_values.sql

```diff
@@ -0,0 +1,70 @@
+#standardSQL
+# Popularity of CSS contain values
+CREATE TEMPORARY FUNCTION getContaine(css STRING)
+RETURNS ARRAY<STRING> LANGUAGE js
+OPTIONS (library = "gs://httparchive/lib/css-utils.js") AS '''
+try {
+  var reduceValues = (values, rule) => {
+    if ("rules" in rule) {
+      return rule.rules.reduce(reduceValues, values);
+    }
+    if (!("declarations" in rule)) {
+      return values;
+    }
+
+    var contain = rule.declarations.find(
+      (d) => d.property == "contain"
+    );
+
+    if (!contain) {
+      return values;
+    }
+
+    return values.concat(contain.value.replace("!important", "").split(" ").filter(Boolean));
```

Comment:
@kevinfarrugia @bkardell should we split the values of the `contain` property? Or should we preserve the entire value?

For example if we're looking for values of `layout inline-size` for [container queries](https://css-tricks.com/next-gen-css-container/) we would only get instances of `layout` and instances of `inline-size` separately, but no stats on when they're used together and in order.

Reply:
I would keep them together -- they are easy enough to see that that is both but impossible to reassmble

Reply:
Updated the query and [results](https://docs.google.com/spreadsheets/d/1w3GzzTNeKxafFODmjDs6OC2dseNEDDKwUV8KeSgRI1Y/edit#gid=1251142331)